examples/ipycanvas.ipynb

```diff
@@ -0,0 +1,150 @@
+    "    \"\"\"Game of life step\"\"\"\n",
+    "    nbrs_count = sum(np.roll(np.roll(x, i, 0), j, 1)\n",
+    "                     for i in (-1, 0, 1) for j in (-1, 0, 1)\n",
+    "                     if (i != 0 or j != 0))\n",
+    "    return (nbrs_count == 3) | (x & (nbrs_count == 2))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def draw(x, canvas, color='black'):\n",
+    "    with hold_canvas(canvas):\n",
+    "        canvas.clear()\n",
+    "        canvas.fill_style = '#FFF0C9'\n",
+    "        canvas.stroke_style = 'white'\n",
+    "        canvas.fill_rect(0, 0, canvas.size[0], canvas.size[1])\n",
+    "\n",
+    "        canvas.fill_style = color\n",
+    "        canvas.stroke_style = color\n",
+    "\n",
+    "        living_cells = np.where(x)\n",
+    "        \n",
+    "        rects_x = living_cells[0] * n_pixels\n",
+    "        rects_y = living_cells[1] * n_pixels\n",
+    "\n",
+    "        canvas.fill_rects(rects_x, rects_y, n_pixels)\n",
+    "        canvas.stroke_rects(rects_x, rects_y, n_pixels)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "glider_gun =\\\n",
+    "[[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1],\n",
+    " [0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,1,1],\n",
+    " [1,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [1,1,0,0,0,0,0,0,0,0,1,0,0,0,1,0,1,1,0,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [0,0,0,0,0,0,0,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\n",
+    " [0,0,0,0,0,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]\n",
+    "\n",
+    "x = np.zeros((50, 70), dtype=bool)\n",
+    "x[1:10,1:37] = glider_gun"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_pixels = 15\n",
+    "\n",
+    "canvas = RoughCanvas(width=x.shape[1]*n_pixels, height=x.shape[0]*n_pixels)\n",
+    "canvas.fill_style = '#FFF0C9'\n",
+    "canvas.stroke_style = 'white'\n",
+    "canvas.fill_rect(0, 0, canvas.size[0], canvas.size[1])\n",
+    "\n",
+    "canvas"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "draw(x, canvas, '#5770B3')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for _ in range(1_000):\n",
+    "    x = life_step(x)\n",
+    "    draw(x, canvas, '#5770B3')\n",
+    "\n",
+    "    sleep(0.1)"
```

Comment:
Maybe we could event use `asyncio` here:

```python
import asyncio
await asyncio.sleep(0.1)
```

https://user-images.githubusercontent.com/591645/122605412-b3227e80-d077-11eb-8c1d-4ec9495dacfe.mp4